jscomp/bin/bsb.ml

```diff
@@ -8510,27 +8514,33 @@ let () =
     end
   else
     begin
-      match Ext_array.find_and_split Sys.argv Ext_string.equal "--" with
+      match Ext_array.find_and_split Sys.argv Ext_string.equal separator with
       | `No_split
         ->
         begin
           Arg.parse bsb_main_flags annoymous usage;
           (* [-make-world] should never be combined with [-package-specs] *)
-          if !make_world then begin
-            (* don't regenerate files when we only run [bsb -clean-world] *)
-            let deps = regenerate_ninja cwd bsc_dir !force_regenerate in
-            make_world_deps deps
+          begin match !make_world, !force_regenerate with 
+            | false, false -> ()
+            | make_world, force_regenerate -> 
+              (* don't regenerate files when we only run [bsb -clean-world] *)
+              let deps = regenerate_ninja cwd bsc_dir force_regenerate in
+              if make_world then begin
+                make_world_deps deps
+              end;
           end;
           if !watch_mode then begin
             watch ()
             (* ninja is not triggered in this case *)
+          end else if not !no_build then begin 
+            ninja_command ninja [||]  
```

Comment:
convinced by @chenglou , -make-world will build it self too, add a `-no-build` option for power users to debug